SampleBrowser/SampleBrowser/UI/Browser/BrowserViewController.swift

```diff
@@ -114,6 +114,25 @@ class BrowserViewController: UIViewController,
         engineSession.findInPageDone()
     }
 
+    func switchToStrictTrackingProtection() {
+        var settings = engineSession.contentBlockingSettings
+        settings.remove(.standard)
+        settings.insert(.strict)
+        engineSession.contentBlockingSettings = settings
+    }
+
+    func switchToStandardTrackingProtection() {
+        var settings = engineSession.contentBlockingSettings
+        settings.remove(.strict)
+        settings.insert(.standard)
```

Comment:
Just wondering here if we should instead just pass the "wanted content blocking mode" to the session, and let the session handle it itself rather than having it public on the engine session?

I think I understand it was done this way due to the "no image mode" right? For context no image mode is only part of the Firefox app, and not Focus 🤔 I know it's currently tied in the "ContentBlocker" code, but I wonder if can/should put "no image mode" as it's own thing (not sure if that's even possible?)

Reply:
>  I wonder if can/should put "no image mode" as it's own thing (not sure if that's even possible?)

Yeah I had debated on how exactly this should work. The reason I had grouped it in along with the strict/standard options was because at least for WebKit all of this is handled in a similar way (the image blocking is just another Content Blocking rule that we apply in addition to the other ones for strict or standard, AFAICS). Though the API doesn't necessarily need to reflect that, it seemed to make sense that the No Image content blocking should be a part of the general `contentBlockingSettings`. But happy to change that if you think it should work differently.

> Just wondering here if we should instead just pass the "wanted content blocking mode" to the session, and let the session handle it itself rather than having it public on the engine session?

I'm not 100% sure if I'm understanding, might need you to clarify this one further. The client-facing protocols I had tried to simplify somewhat to avoid actually passing in the explicit options (which is why the sample browser protocols are just calling e.g. `switchToStrictTrackingProtection`). But I can rework/revisit any of this, LMK.

Reply:
@lmarceau Pushed up refactors to this per our chat, to move the actual updates to the `contentBlockingSettings` privately into WKEngineSession. LMK if that addresses the remaining concerns.